fdbserver/fdbserver.actor.cpp

```diff
@@ -1653,6 +1666,59 @@ struct CLIOptions {
 };
 } // namespace
 
+struct RemoteIKVSServerInterface {
+	constexpr static FileIdentifier file_identifier = 1938843;
+	RequestStream<struct GetRemoteIKVSServerRequest> getRemoteIKVSServerInterf;
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, getRemoteIKVSServerInterf);
+	}
+};
+
+struct GetRemoteIKVSServerRequest {
+	constexpr static FileIdentifier file_identifier = 9393921;
+	ReplyPromise<RemoteIKVSServerInterface> reply;
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, reply);
+	}
+};
+
+ACTOR Future<Void> runRemoteIKVSServer() {
+
+	state RemoteIKVSServerInterface remoteIKVSServer;
+	remoteIKVSServer.getRemoteIKVSServerInterf.makeWellKnownEndpoint(WLTOKEN_FIRST_AVAILABLE,
+	                                                                 TaskPriority::DefaultEndpoint);
+	loop choose {
+		when(wait(delay(1.5))) { std::cout << "Waiting for connection...\n"; }
+		when(GetRemoteIKVSServerRequest req = waitNext(remoteIKVSServer.getRemoteIKVSServerInterf.getFuture())) {
+			std::cout << "received response for running remote IKVS server\n";
+			req.reply.send(remoteIKVSServer);
+		}
+	}
+}
+
+ACTOR Future<Void> testRemoteIKVSClient() {
+	state RemoteIKVSServerInterface server;
+	state NetworkAddress addr = NetworkAddress::parse("127.0.0.1:6666");
+	wait(delay(0.5));
+	std::cout << "initializing connection to server...\n";
+	server.getRemoteIKVSServerInterf = RequestStream<GetRemoteIKVSServerRequest>(Endpoint({ addr }, UID(-1, 2)));
+	RemoteIKVSServerInterface interf = wait(server.getRemoteIKVSServerInterf.getReply(GetRemoteIKVSServerRequest()));
+	std::cout << "Connection established\n";
+	server = interf;
+	return Void();
+}
+
+ACTOR Future<Void> spawnRemoteIKVS() {
+	state ExecCmdValueString snapArg(LiteralStringRef("bin/fdbserver -r remoteIKVS --public_address 127.0.0.1:6666"));
```

Comment:
Please rename `snapArg` to something more descriptive. Also, instead of `LiteralStringRef` you should use [this](https://github.com/apple/foundationdb/blob/49d196e471e995f563f2e178e7417153602bc8e9/flow/Arena.h#L676). So above line should probably something like:

```suggestion
	state ExecCmdValueString remoteIKVSArgs("bin/fdbserver -r remoteIKVS --public_address 127.0.0.1:6666"_sr);
```